source/auth/tests/connection-string.json

```diff
@@ -0,0 +1,291 @@
+{
+	"tests": [
+		{
+			"description": "should use the default source and mechanism",
+			"uri": "mongodb://user:password@localhost",
+			"hosts": [],
+			"valid": true,
+			"warning": false,
+			"auth": {
+				"db": "admin"
+			},
+			"options": null
+		},
+		{
+			"description": "should use the database when no authSource is specified",
+			"uri": "mongodb://user:password@localhost/foo",
+			"hosts": [],
+			"valid": true,
+			"warning": false,
+			"auth": {
+				"db": "foo"
+			},
+			"options": null
+		},
+		{
+			"description": "should use the authSource when specified",
+			"uri": "mongodb://user:password@localhost/foo?authSource=bar",
+			"hosts": [],
+			"valid": true,
+			"warning": false,
+			"auth": {
+				"db": "bar"
+			},
+			"options": null
+		},
+		{
+			"description": "should recognize an empty password",
+			"uri": "mongodb://user:@localhost",
+			"hosts": [],
+			"valid": true,
+			"warning": false,
+			"auth": {
+				"username": "user",
+				"password": ""
+			},
+			"options": null
+		},
+		{
+			"description": "should recognize no password",
+			"uri": "mongodb://user@localhost",
+			"hosts": [],
+			"valid": true,
+			"warning": false,
+			"auth": {
+				"username": "user",
+				"password": null
+			},
+			"options": null
+		},
+		{
+			"description": "should recognize no password",
```

Comment:
This is a duplicate of the test above it.

Reply:
Thank. I noticed that in the .js file too, but forgot to remove it! Fixed now